example_test.go

```diff
@@ -772,3 +772,39 @@ func ExamplePullOpt() {
 		msg.Ack()
 	}
 }
+
+func ExampleContext() {
+	nc, err := nats.Connect("localhost")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	js, _ := nc.JetStream()
+
+	// Base context
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// nats.Context option implements context.Context interface, so can be used
+	// to create a new context from top level one.
+	nctx := nats.Context(ctx)
+
+	// JetStreamManager functions all can use context.
+	js.AddStream(&nats.StreamConfig{
+		Name:     "FOO",
+		Subjects: []string{"foo"},
+	}, nctx)
+
+	// Custom context with timeout
+	tctx, _ := context.WithTimeout(nctx, 2*time.Second)
```

Comment:
I think we need to add this, but not sure if we can edit wally's repo. 😅 
```suggestion
	tctx, tcancel := context.WithTimeout(nctx, 2*time.Second)
	defer tcancel()
```